pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -756,7 +808,28 @@ func isGvk(obj *unstructured.Unstructured, expectGvk *schema.GroupVersionKind) e
 	return nil
 }
 
-func suiteNeedsUpdate(have, found *compliancev1alpha1.ComplianceSuite) bool {
+func (r *ReconcileScanSettingBinding) suiteNeedsUpdate(have, found *compliancev1alpha1.ComplianceSuite, instance *compliancev1alpha1.ScanSettingBinding) bool {
+	// When a ComplianceSuite is created, the settings from the ScanSetting
+	// in the ScanSettingBinding are copied to the ComplianceSuite. This
+	// code checks to make sure any changes to the ScanSetting are
+	// propagated to the ComplianceSuite.
+	scanSetting := &compliancev1alpha1.ScanSetting{}
+	var err error
+	if instance.SettingsRef != nil {
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: instance.SettingsRef.Name, Namespace: instance.Namespace}, scanSetting)
+		if err != nil {
+
+		}
+		log.Info("Found ScanSetting", "ScanSetting", scanSetting.Name)
+		if scanSetting.ComplianceSuiteSettings.Suspend != found.Spec.ComplianceSuiteSettings.Suspend {
+			log.Info("Detected change between ScanSetting and ComplianceSuite", "ScanSetting", scanSetting, "ComplianceSuite", found.Name)
+			return true
+		} else if scanSetting.ComplianceSuiteSettings.Schedule != found.Spec.ComplianceSuiteSettings.Schedule {
+			log.Info("Detected change between ScanSetting and ComplianceSuite", "ScanSetting", scanSetting, "ComplianceSuite", found.Name)
+			return true
+		}
+	}
```

Comment:
I wonder if we need this because, we have 
https://github.com/ComplianceAsCode/compliance-operator/blob/cd3744f61a483d2d04fd79c3eac28e0d026005de/pkg/controller/scansettingbinding/scansettingbinding_controller.go#L339,
the suite should already have expected scansetthing in the have.Spec.ComplianceSuiteSettings

Reply:
I expanded the conditional here to include checking for the scansetting changes because without it the CronJob isn't updated because the it's not detected (line 314). Checking this again though.

Reply:
I was able to revert this and the tests passed, so the new scan settings are getting copied here.